demo-app-benchmark-test/src/main/java/cash/z/ecc/android/sdk/demoapp/benchmark/StartupBenchmark.kt

```diff
@@ -0,0 +1,111 @@
+    }
+
+    @get:Rule
+    val benchmarkRule = MacrobenchmarkRule()
+
+    /**
+     * This test starts the Demo-app on Home screen and measures its metrics.
+     */
+    @Test
+    fun appStartup() = benchmarkRule.measureRepeated(
+        packageName = APP_TARGET_PACKAGE_NAME,
+        metrics = listOf(StartupTimingMetric()),
+        iterations = 5,
+        startupMode = StartupMode.COLD,
+        setupBlock = {
+            // Press home button before each run to ensure the starting activity isn't visible
+            pressHome()
+        }
+    ) {
+        startActivityAndWait()
+    }
+
+    /**
+     * Advanced trace events startup test, which starts the Demo-app on the Home screen and then navigates to the
+     * Get Address screen. Logic for providing addresses from SDK is measured here.
+     */
+    @Test
+    @OptIn(ExperimentalMetricApi::class)
+    fun tracesSdkStartup() = benchmarkRule.measureRepeated(
+        packageName = APP_TARGET_PACKAGE_NAME,
+        metrics = listOf(
+            TraceSectionMetric(sectionName = ADDRESS_SCREEN_SECTION, mode = TraceSectionMetric.Mode.First),
+            TraceSectionMetric(sectionName = UNIFIED_ADDRESS_SECTION, mode = TraceSectionMetric.Mode.First),
+            TraceSectionMetric(sectionName = SAPLING_ADDRESS_SECTION, mode = TraceSectionMetric.Mode.First),
+            TraceSectionMetric(sectionName = TRANSPARENT_ADDRESS_SECTION, mode = TraceSectionMetric.Mode.First)
+        ),
+        compilationMode = CompilationMode.Full(),
+        startupMode = StartupMode.COLD,
+        iterations = 5,
+        measureBlock = {
+            startActivityAndWait()
+            gotoAddressScreen()
+            waitForAddressScreen()
+            closeAddressScreen()
+        }
+    )
+
+    private fun MacrobenchmarkScope.closeAddressScreen() {
+        // To close the Address screen and disconnect from SDK Synchronizer
+        device.pressBack()
+    }
+
+    private fun MacrobenchmarkScope.waitForAddressScreen() {
+        val timeoutSeconds = 5.seconds
+        check(
+            device.run {
+                val ua = Until.hasObject(By.text("^[a-z0-9]{141}$".toPattern())) // NON-NLS (Unified address condition)
+                val sa = Until.hasObject(By.text("^[a-z0-9]{78}$".toPattern())) // NON-NLS (Sapling address condition)
+                val ta = Until.hasObject(By.text("^[a-zA-Z0-9]{35}$".toPattern())) // NON-NLS (Transparent address)
+                // condition
+
+                wait(ua, timeoutSeconds.inWholeMilliseconds) != null &&
+                    wait(sa, timeoutSeconds.inWholeMilliseconds) &&
+                    wait(ta, timeoutSeconds.inWholeMilliseconds)
+            }
+        ) {
+            "Some of the addresses didn't show before $timeoutSeconds seconds timeout."
+        }
+    }
+
+    private fun MacrobenchmarkScope.gotoAddressScreen() {
+        // Open drawer menu
+        device.findObject(By.desc("Open navigation drawer"))    // NON-NLS
```

Comment:
Should we file a followup issue to refactor the demo app to have `demo-ui-lib` so that we can access the string resources programmatically?